src/services/ReportsAPI.js

```diff
@@ -26,15 +56,143 @@ class ReportsAPI {
-            const response = await axios.get(REPORTS_API_URL + '/reports/' + userId)
-            return response.data
+            const response = await axios({
+                url: REPORTS_API_URL + '/reports/' + reportId,
+                headers: {
+                    'Authorization': `Bearer ${token}`
+                },
+                method: 'get'
+            });
+            const report = response.data;
+            return {
+                ...report,
+                comorbidity: report.diagnosedWith,
+                city: report.postalCode
+            }
         } catch (error) {
             console.error(error)
             return []
         }
     }
+
+    static getResultsToDisplay(report) {
+        let resultsToDisplay = {...report};
+
+        resultsToDisplay.sex = this.getSex(report);
+
+        resultsToDisplay.isPregnant = report.isPregnant ? "SÍ" : "NO";
+
+        resultsToDisplay.hasBeenInContactWithInfected = report.hasBeenInContactWithInfected ? "SÍ" : "NO";
+
+        resultsToDisplay.hasBeenTested = report.hasBeenTested ? "SÍ" : "NO";
+
+        resultsToDisplay.testResult = this.getTestResult(report);
+
+        resultsToDisplay.symptoms = this.getSymptoms(report);
+
+        resultsToDisplay.bodyTemperature = report.bodyTemperature !== '' ? report.bodyTemperature + " ºC" : "N/A";
+
+        resultsToDisplay.diagnosedWith = this.getComorbidities(report);
+
+        resultsToDisplay.smokingHabit = report.smokingHabit === "CURRENTLY" ? "SÍ" : "NO";
+
+        resultsToDisplay.isolationStatus = this.getIsolationReason(report);
+
+        resultsToDisplay.submissionTimestamp = this.getSubmissionDate(report);
+
+        return resultsToDisplay;
+    }
+
+    static getSex(report) {
+        let sex = null;
+        if (report.sex === "MALE") {
+            sex = "MASCULINO";
+        }
+        else if (report.sex === "FEMALE") {
+            sex = "FEMENINO";
+        }
+        else {
+            sex = "OTRO";
+        }
+        return sex;
+    }
+
+    static getTestResult(report) {
+        let testResult = null;
+        if (report.testResult === null) {
+            testResult = "N/A";
+        }
+        else if (report.testResult === "POSITIVE") {
+            testResult = "POSITIVO";
+        }
+        else if (report.testResult === "NEGATIVE") {
+            testResult = "NEGATIVO";
+        }
+        else {
+           testResult = "PENDIENTE";
+        }
+        return testResult;
+    }
+
+    static getSymptoms(report) {
+        const symptoms = {...report.symptoms};
+        let symptomsList = new Array();
+        for (let symp in symptoms) {
+            if (symptoms[symp]) {
+                symptomsList.push(SYMPTOMS_MAP[symp]);
+            }
+        }
+        if (symptomsList.length === 0) {
+            symptomsList.push("N/A");
+        }
+        return symptomsList;
+    }
+
+    static getComorbidities(report) {
+        const diagnosedWith = {...report.diagnosedWith};
```

Comment:
list is not being modified so a new object is not required

Reply:
Correct. Same object used.